prisma/schema.prisma

```diff
@@ -39,26 +39,64 @@ model Course {
   createdAt     DateTime       @default(now())
   updatedAt     DateTime       @updatedAt
   authorId      String
+  slug          String         @unique
   name          String
   description   String
   price         Int
   image         String
   category      String
   author        User           @relation(fields: [authorId], references: [id], onDelete: Cascade)
   subscriptions Subscription[]
+  Transaction   Transaction[]
+  Content       Content[]
 }
 
 model Transaction {
   id                String       @id @default(uuid())
   createdAt         DateTime     @default(now())
   updatedAt         DateTime     @updatedAt
   userId            String
+  courseId          String
+  authorId          String
   subscriptionId    String
   bankName          String
   bankAccountNumber String
   bankAccountName   String
   amount            Int
+  dateTime          DateTime
+  method            String
   status            String
   user              User         @relation(fields: [userId], references: [id], onDelete: Cascade)
+  course            Course       @relation(fields: [courseId], references: [id], onDelete: Cascade)
+  author            User         @relation(fields: [authorId], references: [id], onDelete: Cascade)
   subscription      Subscription @relation(fields: [subscriptionId], references: [id], onDelete: Cascade)
 }
+
+model Content {
+  id          String   @id @default(uuid())
+  createdAt   DateTime @default(now())
+  updatedAt   DateTime @updatedAt
+  authorId    String
+  courseId    String
+  slug        String
+  name        String
+  description String
+  type        String
+  content     String
+  author      User     @relation(fields: [authorId], references: [id], onDelete: Cascade)
+  course      Course   @relation(fields: [courseId], references: [id], onDelete: Cascade)
+}
+
+model Consumption {
+  id        String   @id @default(uuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+  userId    String
+  contentId String
+  courseId  String
+  progress  String
```

Comment:
```suggestion
  progress  Int
```
Please update the `progress` field type into `Int`, as expressed in the ERD. Thanks!

Reply:
I'd like to try it. Thank you for the explanation, Mas @zainfathoni !😁🙏

Reply:
Mas Zain @zainfathoni, i have tried searching about the use of denormalization, but i cant find out any examples of implementation of it on creating model in prisma schema. Could you please show me one?😅

Reply:
Denormalization is a database concept, it's not a Prisma-specific use case. There's nothing we need to do on the Prisma side in this PR, we need to do it on the application level later in another PR when we're implementing the logic. 😉

You can read about it [here](https://www.prisma.io/dataguide/intro/database-glossary) and [here](https://www.geeksforgeeks.org/denormalization-in-databases/).

Reply:
Okay, thank you for the explanation and reference, Mas @zainfathoni !

Reply:
I have tried added name argument to the relation since I got validation error when trying to generate a migration file. Please for the correction, Mas @zainfathoni, and sorry for the blocking I caused 🙏

Reply:
I'll take another look later today. No worries! 😉